network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/config/NetPDefaultConfigProvider.kt

```diff
@@ -62,14 +68,45 @@ class RealNetPDefaultConfigProvider @Inject constructor(
     }
 
     override suspend fun routes(): Map<String, Int> = withContext(dispatcherProvider.io()) {
+        val isLocalDnsEnabled = vpnRemoteFeatures.runtimeIpRouteGeneration().isEnabled()
+
         return@withContext if (netPSettingsLocalConfig.vpnExcludeLocalNetworkRoutes().isEnabled()) {
-            WgVpnRoutes.wgVpnDefaultRoutes.toMutableMap().apply {
+            val routes = if (isLocalDnsEnabled) {
+                // get the controller IPs
+                val controllerIPs = vpnLocalDns.lookup(CONTROLLER_NETP_DUCKDUCKGO_COM)
+                    .filterIsInstance<Inet4Address>()
+                    .mapNotNull { it.hostAddress }
+                    .associateWith { 32 }
+
+                val excludedRanges = WgVpnRoutes.vpnDefaultExcludedRoutes + controllerIPs
+                logcat { "Generating VPN routes dynamically, excluded ranges: $excludedRanges" }
+
+                WgVpnRoutes().generateVpnRoutes(excludedRanges)
+            } else {
+                WgVpnRoutes.wgVpnDefaultRoutes
```

Comment:
Why are we keeping this and not using WgVpnRoutes().generateVpnRoutes(WgVpnRoutes.vpnDefaultExcludedRoutes)?

Reply:
I think if we do this, we can get rid of the hardcoded list altogether (wgVpnDefaultRoutes and wgVpnRoutesIncludingLocal). Or are you intentionally keeping them for now for safer fallback for the killswitch?